creating_packages/define_abi_compatibility.rst

```diff
@@ -701,6 +701,49 @@ generated with the ``recipe_revision_mode`` can be resolved if no package for th
             self.compatible_packages.append(p)
 
 
+.. _full_transitive_package_id:
+
+Enabling full transitivity in package_id modes
+++++++++++++++++++++++++++++++++++++++++++++++
+
+When a package declares in its ``package_id()`` method that it is not affected by its dependencies, that will propagate down
+to the indirect consumers of that header only package. If the header only package dependency is a static library, for example, then downstream
```

Comment:
As this affects only to a graph with packages declared as `header_only`, we should start this section with a big warning to make clear that difference, the intention and the rationale: it works this way for all the packages but for header-only ones. From that starting point we can start writing

Reply:
No, it can also affect users calling ``self.info.requires.clear()``, ``self.info.requires.remove["dep"]``, ``self.info.requires.unrelated_mode()``. It is not only for header-only libraries. I will add these cases to the docs.